tests/pyanaconda_tests/iutil_test.py

```diff
@@ -807,3 +808,12 @@ def open_with_perm_test(self):
                 os.umask(old_umask)
         finally:
             shutil.rmtree(test_dir)
+
+class EncryptPasswordTests(unittest.TestCase):
+    def encrypt_password_test(self):
+        """ Test the encrypt_password function"""
+        for algo in ["$1$", "$5$", "$6$"]:
+            enc_pw = iutil.encrypt_password("DocBrown", algo, 16)
+            self.assertEqual(algo, enc_pw[:3])
+            self.assertEqual(crypt.crypt("DocBrown", enc_pw), enc_pw)
+            self.assertNotEqual(crypt.crypt("Einstein", enc_pw), enc_pw)
```

Comment:
Looking at this makes me wonder why we are not using ``crypt.crypt()`` instead of our custom code? Looks like it even got some improvements with Python 3 which were backported to 2.7.